src/nvenc/nvenc_base.cpp

```diff
@@ -457,33 +458,31 @@ namespace nvenc {
   nvenc_base::invalidate_ref_frames(uint64_t first_frame, uint64_t last_frame) {
     if (!encoder || !encoder_params.rfi) return false;
 
-    if (last_frame < first_frame ||
-        encoder_state.last_encoded_frame_index < first_frame ||
-        encoder_state.last_encoded_frame_index > first_frame + 100) {
-      BOOST_LOG(error) << "NvEnc: rfi request " << first_frame << "-" << last_frame << " invalid range (last encoded frame " << encoder_state.last_encoded_frame_index << ")";
-      return false;
-    }
-
-    if (first_frame >= encoder_state.last_rfi_range.first &&
+    if (first_frame == encoder_state.last_rfi_range.first &&

```

Comment:
Is the change from `>=` to `==` correct? If you receive an RFI range of [5, 8] then [6, 8], the latter is a subset of the former, so it should be a no-op.

Reply:
> If you receive an RFI range of [5, 8] then [6, 8]

This shouldn't happen. Until client receives uncorrupted rfi confirmation (or idr) frame, they will continue sending expanding rfi range. So I thought == reflects the behavior more closely, and it won't break even if the client starts sending narrowing range for whatever reason.

But reverted to >=, if the client wants to be stupid, it's on them.